File: leetcode/EasyInterviewQuestions/LinkedLists/mergeSortedLists.js
/**
 * Merge two sorted linked lists and return it as a new list. The new list should be made by splicing together the nodes of the first two lists.

Example:

Input: 1->2->4, 1->3->4
Output: 1->1->2->3->4->4
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} l1
 * @param {ListNode} l2
 * @return {ListNode}
 */
var mergeTwoLists = function(l1, l2) {
  if (!l1) {
    return l2;
  }
  if (!l2) {
    return l1;
  }
  var head = null;
  if (l1.val <= l2.val) {
    head = l1;
    l1 = l1.next;
  } else {
    head = l2;
    l2 = l2.next;
  }
  let p = head;
  while (l1 && l2) {
    if (l1.val <= l2.val) {
      p.next = l1;
      l1 = l1.next;
    } else {
      p.next = l2;
      l2 = l2.next;
    }
    p = p.next;
  }
  if (l1) {
    p.next = l1;
  } else {
    p.next = l2;
  }
  return head;
};

File: Two Pointers/mergeSortedLists.py


def merge(A, B):
    m = len(A)
    n = len(B)
    i = 0
    j = 0
    ans = []
    while(i < m and j < n):
        if(A[i] < B[j]):
            ans.append(A[i])
            i += 1
        else:
            ans.append(B[j])
            j+=1
    return ans + A[i:] + B[j:]

a = [1, 5, 8]
b = [6, 9]
print(merge(a,b))